src/ModifyQueryArgs.php

```diff
@@ -60,8 +61,9 @@ public function restBasesToPostTypeSlugs(array $postTypes): array
 	{
 		$postTypeSlugs = [];
 		foreach ($postTypes as $postTypeRestBase) {
-			if ($this->preparedPostTypes->restBaseToSlug($postTypeRestBase)) {
-				$postTypeSlugs[] = $this->preparedPostTypes->restBaseToSlug($postTypeRestBase);
+			$newSlug = $this->preparedPostTypes->restBaseToSlug($postTypeRestBase);
+			if (!empty($newSlug)) {
```

Comment:
This is better, but my concern is if the conditional isn't met, then the results array has the wrong count. Maybe it should throw an exception if it can't create a slug, as that should always be possible.

Reply:
>but my concern is if the conditional isn't met, then the results array has the wrong count. 

This concern exists now in your code. The above proposed change merely captures the result and prevents calling the method twice.  It's more performant.  

But it does not mitigate any count concerns when a post type does not have a rest base.  That does need to be handled.  

How? You could throw an error, as that's the easiest solution.  It provides clear feedback to the developer that something went wrong and we can't process this particular post type.

The question then becomes: is this a recoverable error condition, i.e. can we continue processing the rest of the post types to provide search results back to the viewer?